eduNEXT/openedx-filters
# feat: add class based tooling for filters

**Description:** 
This PR  adds class-based tooling for filter execution.

**Dependencies:** 
None for now.

**Merge deadline:** 
None.

**Installation instructions:** 
Install openedx-filters: `pip install git+https://github.com/eduNEXT/openedx-filters.git@MJG/class_filter_tooling#egg=openedx_filters==0.2.0_alpha`

**Testing instructions:**
1. In your edx-platform version, you can check out to this branch or add the changes yourself:
https://github.com/edx/edx-platform/pull/28773
2. Change your settings according to your test plugin and add the following:
```
from openedx_filters import PipelineStep
from openedx_filters.learning.enrollment import PreEnrollmentFilter

class CanPerformEnrollment(PipelineStep):
    """
    Filter step used as pre-enrollment approval.
    """

    def do_filter(self, user, course_key, mode):
       "Method that raises exception used to avoid enrollments."
        raise PreEnrollmentFilter.PreventEnrollment
```
3. Enroll in a  course

**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** 
None for now
